{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enter userhandle: sumit_011\n",
      "\n",
      "\n",
      "Fully Solved (124)\n",
      "Practice: RIGHTRI JDELAY KTTABLE STRSRCH LECANDY CNOTE INF1804 ADASTAIR BINIM2 SALARY KCHESS SNCKYEAR CHFAR TRUEDARE THREEFR COPS FLOW018 FLOW008 CLFIBD ZUBTRCNT PLAYPIAN NOMATCH SOPC18A CHCOINSG PPATTERN AGECAL SNAPE COOLING MAXCOUNT NUMGAME DCE05 PRPALIN BUY1GET1 SPALNUM ADASCOOL ADAKNG DEVARRAY LONGSEQ CLIPLX VALIDSTK MGCSET CHEFSIGN BINTREE QUALPREL CHFIDEAL PCJ18C ISITCAKE CHFINTRO COINS FLOW002 TSORT FLOW006 FCTRL2 START01 FLOW001 FLOW004 NBONACCI LUCKFOUR FLOW007 FSQRT FLOW017 FLOW016 PALL01 OJUMPS TRICOIN PRF8 POGOSTCK ALTARAY XORAGN GCDSET FLOW010 CV DPS RECVSEQ DANOW RRSTONE CSUB FIRESC PLAYSTR TWOVRIBL CW1001 \n",
      "SEPT18B: MAGICHF CHEFADV \n",
      "CGSC2018: CGAAST \n",
      "COOK98B: ATM2 MAKPERM \n",
      "OCT18B: CHSERVE BITOBYT MINDSUM HMAPPY CCIRCLES \n",
      "NOV18B: CHFTIRED PRDRG \n",
      "GWR18ROL: LIFELTD \n",
      "JAN19B: FANCY MGAME DIFNEIGH \n",
      "FEB19B: HMAPPY2 CHEFING DEPCHEF MAGICJAR \n",
      "LTIME69B: AVG TRICKYDL \n",
      "MARCH19B: CHDIGER CHNUM \n",
      "CORU2019: ROBRESC \n",
      "BION2019: VWL \n",
      "ENMA2019: NUTLOOP AVH05 \n",
      "LTIME70B: TSTROBOT \n",
      "APRIL19B: MAXREM STRCH FENCE \n",
      "MAY19B: REDONE \n",
      "COOK106B: REACTION \n",
      "JULY19B: CHFM MMAX \n",
      "LTIME74B: RPD \n",
      "AUG19B: CHEFDIL ZOMCAV MSNSADM1 DSTAPLS KS1 \n",
      "\n",
      "\n",
      "Partially Solved (8)\n",
      "SEPT18B: XORIER ANDSQR \n",
      "CGSC2018: CGSURYA \n",
      "NOV18B: CHHAPPY HMAPPY1 \n",
      "JAN19B: HP18 \n",
      "JULY19B: PRTAGN \n",
      "AUG19B: ENCODING \n",
      "\n",
      "\n",
      "content found\n",
      "Enter problem code: REDONE\n"
     ]
    },
    {
     "ename": "WebDriverException",
     "evalue": "Message: 'chromedriver' executable needs to be in PATH. Please see https://sites.google.com/a/chromium.org/chromedriver/home\n",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/selenium/webdriver/common/service.py\u001b[0m in \u001b[0;36mstart\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     75\u001b[0m                                             \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlog_file\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 76\u001b[0;31m                                             stdin=PIPE)\n\u001b[0m\u001b[1;32m     77\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/subprocess.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, encoding, errors, text)\u001b[0m\n\u001b[1;32m    774\u001b[0m                                 \u001b[0merrread\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merrwrite\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 775\u001b[0;31m                                 restore_signals, start_new_session)\n\u001b[0m\u001b[1;32m    776\u001b[0m         \u001b[0;32mexcept\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/subprocess.py\u001b[0m in \u001b[0;36m_execute_child\u001b[0;34m(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, restore_signals, start_new_session)\u001b[0m\n\u001b[1;32m   1521\u001b[0m                             \u001b[0merr_msg\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;34m': '\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mrepr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merr_filename\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1522\u001b[0;31m                     \u001b[0;32mraise\u001b[0m \u001b[0mchild_exception_type\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merrno_num\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merr_msg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merr_filename\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1523\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0mchild_exception_type\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merr_msg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'chromedriver': 'chromedriver'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mWebDriverException\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-0255fc834f36>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     52\u001b[0m \u001b[0mproblem\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Enter problem code: '\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     53\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mproblem\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mlinks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 54\u001b[0;31m     \u001b[0mdriver\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mwebdriver\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mChrome\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     55\u001b[0m     \u001b[0mdriver\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'https://www.codechef.com'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mlinks\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mproblem\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     56\u001b[0m \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/selenium/webdriver/chrome/webdriver.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, executable_path, port, options, service_args, desired_capabilities, service_log_path, chrome_options)\u001b[0m\n\u001b[1;32m     66\u001b[0m             \u001b[0mservice_args\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mservice_args\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     67\u001b[0m             log_path=service_log_path)\n\u001b[0;32m---> 68\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mservice\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstart\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     69\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     70\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/selenium/webdriver/common/service.py\u001b[0m in \u001b[0;36mstart\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     81\u001b[0m                 raise WebDriverException(\n\u001b[1;32m     82\u001b[0m                     \"'%s' executable needs to be in PATH. %s\" % (\n\u001b[0;32m---> 83\u001b[0;31m                         os.path.basename(self.path), self.start_error_message)\n\u001b[0m\u001b[1;32m     84\u001b[0m                 )\n\u001b[1;32m     85\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merrno\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0merrno\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mEACCES\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mWebDriverException\u001b[0m: Message: 'chromedriver' executable needs to be in PATH. Please see https://sites.google.com/a/chromium.org/chromedriver/home\n"
     ]
    }
   ],
   "source": [
    "from selenium import webdriver\n",
    "from bs4 import BeautifulSoup\n",
    "import requests\n",
    "import sys\n",
    "\n",
    "links = {}\n",
    "\n",
    "def display_all_problems(user):\n",
    "    try:\n",
    "        try:\n",
    "            response = requests.get('https://www.codechef.com/users/'+user)\n",
    "        except:\n",
    "            input('response could not be created')\n",
    "            return False\n",
    "        \n",
    "        try:\n",
    "            soup = BeautifulSoup(response.text,'lxml')\n",
    "        except:\n",
    "            input('Could not make soup')\n",
    "            return False\n",
    "\n",
    "        content = soup.find('section',{'class':'rating-data-section problems-solved'})\n",
    "        Solved = content.find_all(['h5'])\n",
    "        sets = content.find_all(['article'])\n",
    "\t\n",
    "        print('\\n')\n",
    "        for i in range(0,2):\n",
    "            print(Solved[i].text)\n",
    "            types = sets[i].find_all(['p'])\n",
    "\t        \n",
    "            for typ in types:\n",
    "                competition = typ.find(['strong']).text\n",
    "                print(competition,end = ' ')\n",
    "                problems = typ.find_all(['a'])\n",
    "                for prob in problems:\n",
    "                    links[prob.text]=prob['href']\n",
    "                    print(prob.text,end = ' ')\n",
    "                print()\n",
    "\t            \n",
    "            print('\\n')\n",
    "\t\n",
    "        input('content found')\n",
    "        return True\n",
    "    except:\n",
    "        input('content not found')\n",
    "        return False\n",
    "\n",
    "\n",
    "usr = input('Enter userhandle: ')\n",
    "if not display_all_problems(usr):\n",
    "    sys.exit()\n",
    "problem = input('Enter problem code: ')\n",
    "if problem in links.keys():\n",
    "    driver = webdriver.Chrome()\n",
    "    driver.get('https://www.codechef.com'+links[problem])\n",
    "else:\n",
    "    input('Problem not done by user')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
